spec-0006/index.md

```diff
@@ -0,0 +1,113 @@
+---
+title: "SPEC 6 — Keys to the Castle"
+date: 2022-12-19
+author:
+  - "Stéfan van der Walt <stefanv@berkeley.edu>"
+  - "Jarrod Millman <millman@berkeley.edu>"
+discussion:
+endorsed-by:
+---
+
+## Description
+
+<!--
+Briefly and clearly describe the proposal.
+Explain the general need and the advantages of this specific proposal.
+If relevant, include examples of how the new functionality would be used,
+intended use-cases, and pseudo-code illustrating its use.
+-->
+
+Projects engage with restricted resources all the time.
+Examples include access to add & remove team members, grant commit rights, or make uploads to certain hosts.
+
+**Documenting resources** is critical to ensuring uninterrupted operations.
+It is important that team members know _who_ have access to resources, and how to _gain access_ to resources.
+For example, if updated project documentation needs to be uploaded to a remote server via SSH, who are the team members that can do that, and by which process can a release manager request access?
+
+Furthermore, project developers sometimes have to **share secrets**, such as server passwords and social media logins.
+Typically, such secrets are distributed among those who need them, without any centralized system for tracking the secrets or who has access to them.
+When a server needs to be accessed, there is often a scramble to find someone with credentials.
+
+This SPEC discusses the requirements for a system to distribute secrets, provides an example implementation, and lists suitable hosted services.
+
+It should be noted that, in many cases, secrets & passwords can be avoided.
+Most online services (GitHub, PyPi, etc.), have permissioning systems through which developers can be given the required access.
+Access to servers can be given through SSH keys, which each developer has to safeguard.
+This document deals with the situation in which that is not possible, and secrets have to be shared.
+
+## Implementation
+
+### Principles
```

Comment:
Nitpick: Feels weird to me that "Principles" is in the section "Implementation". I'd make it its own level-`##` section after "Description" and before "Implementation".